compiler-rt/lib/rtsan/rtsan_interceptors_posix.cpp

```diff
@@ -302,6 +302,15 @@ INTERCEPTOR(int, fpurge, FILE *stream) {
   __rtsan_notify_intercepted_call("fpurge");
   return REAL(fpurge)(stream);
 }
+#define RTSAN_MAYBE_INTERCEPT_FPURGE INTERCEPT_FUNCTION(fpurge)
+#define RTSAN_MAYBE_INTERCEPT_PIPE2
+#else
```

Comment:
I would split this into its own ifdef, I think fpurge and pipe2 are unrelated to one another